### PR TITLE
Added event sliced mergeing support

### LINKED ISSUE #19196: Merged reduction of time sliced event workspaces does not work for ISIS SANS
Sarah reported that merged reductions of time sliced data do not work. The issue appears in the way the merge-mechanism looks for the relevant workspaces. It seems to be looking for workspaces without the time slice specified. 

### Expected behavior

Should merge on a per time slice basis

### Actual behavior

Does not find the reduced data from the front detector and fails.

### Steps to reproduce the behavior

Perform a time sliced merged reduction

### Platforms affected

ALL

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Don't Merge yet want to tidy commits**
The SANS reduction framework was failing when you attempted to do a merged reduction of event sliced data. 

**To test:**
- Get the SANS2D dataset from \\olympic\Babylon5\Scratch\Matthew Andrew\GUI_TEST
- Add the folder to the directories path in mantid.

- Open the old SANS GUI, Interfaces-> SANS
- Select SANS2DTUBES as the instrument
- Select Batch Mode
- Click on load user file navigate to GUI_TEST and load USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_REAR.txt

- Click on load batch file, load batch_file

- In the time slices box enter 1:1:3
- In the reduction settings tab selected merged as the reduction mode (labelled Detector Bank on the old gui).
- Click 1D Reduce
- Confirm that workspace groups called line_1_reduction_front, line_1_reduction_rear and line_1_reduction_merged are produced.

- Open the new sans Gui and repeat the above reduction but change the output name to 'New' and select compatibility mode from the settings->General. Scale, Event Slice, Sample tab.

- Confirm that a workspace group called New_rear is produced (Note this naming seems to be a bug with the new GUI)

- Confirm that the individual workspaces within New_rear and line_1_reduction_merged are the same.

Fixes #19196 . 

**Release Notes** 
Updated release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/ISISCommandInterface.py [hunk 1]
```diff
@@ -494,107 +494,26 @@ def WavRangeReduction(wav_start=None, wav_end=None, full_trans_wav=None, name_su
...
-        retWSname_front = _WavRangeReduction(name_suffix)
+        retWSname_front, front_slices = _WavRangeReduction(name_suffix)
         retWSname = retWSname_front
 
     # This section provides a the REAR -- FRONT fitting and a stitched workspace.
     # If merge_flag is selected we use SANSStitch and get the fitting for free
     # If fitRequired is selected, then we explicity call the SANSFitScale algorithm
     if merge_flag:
-        # Prepare the Norm and Count workspaces for the FRONT and the REAR detectors
-        retWSname_merged = retWSname_rear
```
Comment (gemmaguest) [inline scripts/SANS/ISISCommandInterface.py]: It looks like `retWSname_merged` might still be used further on in the function even though it's no longer being set in this scope?

**Review by gemmaguest — CHANGES_REQUESTED** [review]
The tests behave as described and the output workspaces contain the same contents from both the old and new GUIs :+1:

The names for the `New_rear` workspaces contain a `_merged` suffix whereas for the `line_1_reduction_merged` workspaces they do not. Is this intentional?

Just one in-line query for something I wasn't sure looked right (possibly related to the naming inconsistency?).

File: scripts/SANS/ISISCommandInterface.py (hunk 1 above)
Comment (Mantid-Matthew) [reply scripts/SANS/ISISCommandInterface.py]: I think the naming in the new GUI is wrong here but this is unrelated to these changes which all relate to the old GUI. I've corrected the fact that resWSname_merged was not being used as well.

**Review by gemmaguest — APPROVED** [review]

**Comment by gemmaguest** [comment]
Looks good to me now :+1:

**Pull request merged**